ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala

```diff
@@ -82,9 +82,9 @@ trait LedgerClientBase extends StrictLogging {
       aesf: ExecutionSequencerFactory,
   ): Future[LedgerClientBase.Error \/ DamlLedgerClient] =
     buildLedgerClient(ledgerHost, ledgerPort, clientConfig, nonRepudiationConfig)
-      .map(_.right)
+      .map(\/.r[LedgerClientBase.Error](_))
       .recover { case NonFatal(e) =>
-        \/.left(
+        \/.l[DamlLedgerClient](
```

Comment:
Can I somewhere read up why this was decided? I like types but explicitly typing everywhere while I have type inference is not my strongest preference.

Reply:
Note, take my question with a grain of salt, as my knowledge about this code here is pretty limited :grimacing:

Reply:
Start here https://github.com/digital-asset/daml/pull/9997#discussion_r651998142

Note that this is not necessary if you have context ([expected type](https://discuss.daml.com/t/scala-inference-rules-of-thumb/2242#type-parameters-can-be-inferred-from-expected-type-5), which goes backwards into lambdas). I simply added this where I didn't have context.